timemachine/cpp/src/bd_exchange_move.cu

```diff
@@ -89,24 +96,27 @@ void BDExchangeMove<RealType>::move(
     this->compute_initial_weights(N, d_coords, d_box, stream);
 
     for (int move = 0; move < proposals_per_move_; move++) {
-        if (noise_offset_ >= this->d_quaternions_.length) {
-            // reset the noise to zero and generate more noise
-            noise_offset_ = 0;
-            curandErrchk(templateCurandNormal(cr_rng_, d_quaternions_.data, d_quaternions_.length, 0.0, 1.0));
-        }
         // Run only after the first pass, to maintain meaningful `log_probability_host` values
         if (move > 0) {
             // Run a separate kernel to replace the before log probs and weights with the after if accepted a move
             // Need the weights to sample a value and the log probs are just because they aren't expensive to copy
             k_store_accepted_log_probability<RealType><<<1, tpb, 0>>>(
                 num_target_mols_,
-                d_translations_.data + 3, // Offset to get the last value for the acceptance criteria
+                d_translations_.data + (noise_offset_ * TRANSLATIONS_PER_STEP) +
+                    3, // Offset to get the last value for the acceptance criteria
```

Comment:
initialized to `d_translations_(round_up_even(TRANSLATIONS_PER_STEP * this->RANDOM_BATCH_SIZE))`

Reply:
nm - this is just grabbing the `w` component of `xyzw`